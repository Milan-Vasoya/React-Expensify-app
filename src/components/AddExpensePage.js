
import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { startAddExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.history.push('/expensify');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-containter">
            <h1 className="header__title">Add Expense</h1>
          </div>
        </div>
        <div className="content-containter" >
          <ExpenseForm
            onSubmit={this.onSubmit}
          />

        </div>


      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
