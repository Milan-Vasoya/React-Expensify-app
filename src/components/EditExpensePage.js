import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';


export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/expensify');
    }
    onClick = (e) => {
        this.props.startRemoveExpense(this.props.expense.id);
        this.props.history.push('/expensify');

    }
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-containter">
                        <h1 className="header__title">Edit Expense</h1>
                    </div>
                </div> 
                <div className="content-containter" >
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />

                    <button className='button button--secondary' onClick={this.onClick} > Remove Expense </button>
                </div>
            </div>
        );
    }
}





const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    };
};

const mapDispatchToprops = (dispatch) => {
    return {
        startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
        startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
    }
}

export default connect(mapStateToProps, mapDispatchToprops)(EditExpensePage);