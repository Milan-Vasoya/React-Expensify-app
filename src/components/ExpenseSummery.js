import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';
import totalExpenseAmount from '../selectors/expense-Total';
import numeral from 'numeral';



export const ExpenseListHeader =({expensesCount,totalExpenseAmount})=>{
    const totalformattedAmount=numeral( totalExpenseAmount / 100).format('$0,0.00');
    const expenseWord = expensesCount >= 2 ?'expeses': 'expense';
    
    return(
        <div>
           <h1> viewing {expensesCount} {expenseWord} Total amount: {totalformattedAmount} </h1>
        </div>
    );
};

const mapStateToProps =(state)=>{
    const visibleExpenses=selectExpenses(state.expenses,state.filters);
    return{
        expensesCount:visibleExpenses.length,
        totalExpenseAmount:totalExpenseAmount(visibleExpenses)

    };

}

export default connect(mapStateToProps)(ExpenseListHeader);

