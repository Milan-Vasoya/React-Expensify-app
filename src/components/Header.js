import React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogOut} from '../actions/auth';

export const Header = ({startLogOut}) => (
    <header>
        <h1>Expensify</h1>
        <NavLink to='/dashboard' activeClassName='is-active'  >Go To Home</NavLink>  ||
        <NavLink to='/create' activeClassName='is-active' > Create Page</NavLink> ||    
        {/* <NavLink to='/help' activeClassName='is-active' > Help Page</NavLink> ||   */}
        <button onClick={startLogOut} >Log out</button>
    </header>
);

const mapDispatchToProps= (dispatch)=>({
  startLogOut:()=>dispatch(startLogOut())  
})
export default connect (undefined,mapDispatchToProps) (Header);