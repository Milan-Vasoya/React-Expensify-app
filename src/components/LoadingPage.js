import React from 'react';

const LoadinPage = () =>(
    <div className='loader'>
        <img className='loader--image' src='./img/loader.gif' />
    </div>
)

export default LoadinPage;