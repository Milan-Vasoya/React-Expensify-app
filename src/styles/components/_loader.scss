.loader{
    // text-align: center;
    align-items: center;
    display: flex;
    justify-content: center;
    height: 100vh;
    width: 100vw;
}

.loader--image{
    height: 6rem;
    width: 6rem;
    
}