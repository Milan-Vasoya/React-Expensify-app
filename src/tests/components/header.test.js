import React from 'react';
import {Header} from '../../components/Header';
import {shallow} from 'enzyme';

test('Should render header', ()=>{

    const wrapper=shallow(<Header startLogOut={()=>{}} />);
    expect(wrapper).toMatchSnapshot();

    //  const renderer=new ReactShallowRenderer();
//  renderer.render(<Header />);
//  expect(renderer.getRenderOutput()).toMatchSnapshot();
 
});

test('should simulate button click',()=>{
    const startLogOut= jest.fn();
    const wrapper=shallow(<Header startLogOut={startLogOut} />);
    wrapper.find('button').simulate('click');
    expect(startLogOut).toHaveBeenLastCalledWith();
});