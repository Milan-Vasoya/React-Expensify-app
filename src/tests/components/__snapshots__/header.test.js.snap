// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render header 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/expensify"
  >
    Go To Home
  </NavLink>
    ||
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
     Create Page
  </NavLink>
   ||
  <NavLink
    activeClassName="is-active"
    to="/help"
  >
     Help Page
  </NavLink>
   ||
</header>
`;
